lib/types/number.js

```diff
@@ -122,6 +122,8 @@ module.exports = Any.extend({
         greater: {
             method(limit) {
 
+                Assert(limit !== undefined, 'limit is required');
```

Comment:
The number case is already implemented by the [constraint](https://github.com/hapijs/joi/blob/master/lib/types/number.js#L113-L118) on the limit argument, it's just that the validation isn't run. Moving [this block](https://github.com/hapijs/joi/blob/master/lib/base.js#L591-L594) below the very next `if` seems to solve the problem without breaking other tests, I think it's preferable as it will fix all other rules that are using `assert`. It will also fix the `string` case since the `length` rule is checked.

Reply:
Understood. I made the changes and also adapted the tests.